### PR TITLE
Update code formatting guidelines

### LINKED ISSUE #930: Add code formatting guidelines
See https://github.com/spring-cloud/spring-cloud-dataflow/commit/da2a1bccac25a13aaa090ccf37056af740e96654 for details

* Add Eclipse formatter rules 
* Add README instructions

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Push formatted code

Resolves #930

### CODE REVIEW

File: spring-cloud-stream-tools/src/main/resources/eclipse/eclipse-code-formatter.xml [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+                 value="do not insert"/>
+        <setting id="org.eclipse.jdt.core.formatter.comment.format_header" value="false"/>
+        <setting id="org.eclipse.jdt.core.formatter.alignment_for_arguments_in_allocation_expression" value="16"/>
+        <setting id="org.eclipse.jdt.core.formatter.insert_space_after_opening_paren_in_method_invocation"
+                 value="do not insert"/>
+        <setting id="org.eclipse.jdt.core.formatter.insert_space_before_opening_paren_in_while" value="insert"/>
+        <setting id="org.eclipse.jdt.core.compiler.codegen.inlineJsrBytecode" value="enabled"/>
+        <setting id="org.eclipse.jdt.core.formatter.insert_space_before_closing_paren_in_switch" value="do not insert"/>
+        <setting id="org.eclipse.jdt.core.formatter.alignment_for_method_declaration" value="0"/>
+        <setting id="org.eclipse.jdt.core.formatter.join_wrapped_lines" value="true"/>
```
Comment (mbogoevici) [inline spring-cloud-stream-tools/src/main/resources/eclipse/eclipse-code-formatter.xml]: This should be set to `false` IMO. If the user wants to break down code in a certain way (e.g. for a clearer rendering of fluent APIs) we shouldn't forcibly reformat.

**Comment by mbogoevici** [comment]
As a general comment, I think that fixing https://github.com/spring-cloud/spring-cloud-stream/pull/955#discussion_r116039877 should also be accompanied by the reversal of the formatting changes which are also part of this PR.

File: spring-cloud-stream-tools/src/main/resources/eclipse/eclipse-code-formatter.xml (hunk 1 above)
Comment (mbogoevici) [inline spring-cloud-stream-tools/src/main/resources/eclipse/eclipse-code-formatter.xml]: A similar issue has been opened for https://github.com/spring-cloud/spring-cloud-dataflow/issues/1490

**Comment by ilayaperumalg** [comment]
@mbogoevici Updated the rule to use `join_wrapped_lines` to `false`. But, the reformatting after changing the rule doesn't pick up any changes in the files. Do you see any files that still have incorrect formatting?

**Comment by mbogoevici** [comment]
> But, the reformatting after changing the rule doesn't pick up any changes in the files. 

Not sure I get that. We should just revert the changed files. I suppose I can do that on merge.

**Comment by ilayaperumalg** [comment]
> We should just revert the changed files

Do you mean all the files that are changed as part of this PR? I believe we still need to apply the code formatting changes for the files based on the updated formatting rules.

**Comment by mbogoevici** [comment]
>  I believe we still need to apply the code formatting changes for the files based on the updated formatting rules.

We need to reapply the updated formatting rules on the original files. Reapplying the rules after the first commit will preserve the formatting changes that we don't want (joining the lines).

File: README.adoc [hunk 2]
```diff
@@ -37,6 +37,17 @@ Any question that is not a bug or an issue should be asked on Stack Overflow, us
 
 We love contributions.  Follow this https://github.com/spring-cloud/spring-cloud-commons#contributing[link] for more information on how to contribute.
 
+=== Code formatting guidelines
+
+* The directory `spring-cloud-stream-tools/src/main/resources/eclipse/` has two files for use with code formatting, `eclipse-code-formatter.xml` for the majority of the code formatting rules and `eclipse.importorder` to order the import statements.
+
+* In eclipse you import these files by navigating `Windows -> Preferences` and then the menu items `Preferences > Java > Code Style > Formatter` and `Preferences > Java > Code Style > Organize Imports` respectfully.
```
Comment (mbogoevici) [inline README.adoc]: Capitalize `Eclipse`

File: README.adoc [hunk 3]
```diff
@@ -37,6 +37,17 @@ Any question that is not a bug or an issue should be asked on Stack Overflow, us
 
 We love contributions.  Follow this https://github.com/spring-cloud/spring-cloud-commons#contributing[link] for more information on how to contribute.
 
+=== Code formatting guidelines
+
+* The directory `spring-cloud-stream-tools/src/main/resources/eclipse/` has two files for use with code formatting, `eclipse-code-formatter.xml` for the majority of the code formatting rules and `eclipse.importorder` to order the import statements.
+
+* In eclipse you import these files by navigating `Windows -> Preferences` and then the menu items `Preferences > Java > Code Style > Formatter` and `Preferences > Java > Code Style > Organize Imports` respectfully.
+
+* In `IntelliJ`, install the plugin `Eclipse Code Formatter`.  You can find it by searching the "Browse Repositories" under the plugin option within `IntelliJ` (Once installed you will need to reboot Intellij for it to take effect).
```
Comment (mbogoevici) [inline README.adoc]: One sentence per line.

**Comment by mbogoevici** [comment]
One thing that we should do is to move the `eclipse` folder at the top level. It's easier to point to and the eclipse configuration files don't need to be included in the `tools` jar. WDYT @ilayaperumalg  ?

**Comment by ilayaperumalg** [comment]
@mbogoevici yeah, that's better too. will fix that now.

**Comment by mbogoevici** [comment]
Squashed, rebased and merged as https://github.com/spring-cloud/spring-cloud-stream/commit/bd002e4aaf05f2f4e31cb09d706910b470f960a3

**Pull request closed without merging**